ignek_intranet/modules/common/src/main/java/com/ignek/intranet/common/service/EmployeeServiceImpl.java

```diff
@@ -0,0 +1,212 @@
+package com.ignek.intranet.common.service;
+
+import java.io.ByteArrayOutputStream;
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.ignek.intranet.common.constants.IntranetConstants;
+import com.ignek.intranet.common.employee.response.EmployeeResponse;
+import com.ignek.intranet.common.util.CommonUtil;
+import com.ignek.intranet.employee.model.Employee;
+import com.ignek.intranet.employee.service.EmployeeLocalService;
+import com.itextpdf.text.Document;
+import com.itextpdf.text.DocumentException;
+import com.itextpdf.text.Paragraph;
+import com.itextpdf.text.pdf.PdfWriter;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.search.BooleanClauseOccur;
+import com.liferay.portal.kernel.search.BooleanQuery;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.search.Hits;
+import com.liferay.portal.kernel.search.IndexSearcher;
+import com.liferay.portal.kernel.search.ParseException;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.search.SearchEngineHelper;
+import com.liferay.portal.kernel.search.SearchEngineHelperUtil;
+import com.liferay.portal.kernel.search.SearchException;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.search.generic.BooleanQueryImpl;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+@Component(immediate = true, service = EmployeeService.class)
+public class EmployeeServiceImpl implements EmployeeService {
+
+	@Reference
+	private UserLocalService userLocalService;
+
+	@Reference
+	private RoleLocalService roleLocalService;
+
+	@Reference
+	private EmployeeLocalService employeeLocalService;
+
+	@Reference
+	private SearchEngineHelper searchEngineHelper;
+
+	@Override
+	public EmployeeResponse addUser(long empId, long userId, long companyId, String firstName, String lastName,
```

Comment:
Wrap add/update logic into single method : saveEmployee

Reply:
Or think of adding this logic in service builder, in that case we will not need this additional service for managing employees.